### PR TITLE
[doc] Improve wording on Xprop

### PR DESCRIPTION
**Pull request by sriyerg**
Update Xprop section of the dv methodology doc and the checklist to make it clear what is expected.

Signed-off-by: Srikrishna Iyer <sriyer@google.com>

### CODE REVIEW

**Review by weicaiyang — APPROVED** [review]
LGTM, included the checklist update in the V3 RFC, thanks

File: doc/ug/dv_methodology/index.md [hunk 1]
```diff
@@ -528,6 +528,8 @@ We choose **Pessimistic Mode** as we want to avoid using X value in the conditio
 Xprop is enabled by default when running simulations for all of our DUTs due to the acceptable level of overhead it adds in terms of wall-clock time (less than 10%).
 
 It's mandatory to enable Xprop when running regression for coverage closure.
+Please refer to the simulator documentation to identify and apply the necessary build-time and / or run-time options to enable Xprop.
+On OpenTitan, Xprop is already enabled by default for VCS and Xcelium simulators.
```
Comment (marnovandermaas) [inline doc/ug/dv_methodology/index.md:532]: Nit:
```suggestion
In OpenTitan, Xprop is enabled by default for VCS and Xcelium simulators.
```

File: doc/ug/dv_methodology/index.md [hunk 2]
```diff
@@ -544,6 +546,11 @@ For example, when a_valid is 0 in the TLUL interface, we drive data, address and
   endfunction : invalidate_a_channel
 ```
 
+The simulator may report that some portions of RTL / DV could not instrumented for X-propagation due to the way they were written.
```
Comment (marnovandermaas) [inline doc/ug/dv_methodology/index.md:549]:
```suggestion
The simulator may report that some portions of RTL / DV could not be instrumented for X-propagation due to the way they were written.
```

**Review by marnovandermaas — APPROVED** [review]
LGTM, just a few minor nits.

File: doc/ug/dv_methodology/index.md (hunk 1 above)
Comment (sriyerg) [reply doc/ug/dv_methodology/index.md]: Fixed.

File: doc/ug/dv_methodology/index.md (hunk 2 above)
Comment (sriyerg) [reply doc/ug/dv_methodology/index.md]: Fixed.

**Review by andreaskurth — COMMENTED** [review]
:+1: Thanks for adding this info on Xprop, @sriyerg!

**Pull request merged**